src/NATS.Client.Core/NatsTlsOpts.cs

```diff
@@ -62,7 +63,7 @@ public sealed record NatsTlsOpts
     /// <summary>
     /// Callback that loads Client Certificate
     /// </summary>
-    public Func<ValueTask<X509Certificate2>>? LoadClientCert { get; init; }
+    public Func<ValueTask<X509Certificate2Collection>>? LoadClientCerts { get; init; }
```

Comment:
I don't think we should necessarily pluralize `LoadClientCerts` - it is still just 1 cert plus intermediates.  Plural to me indicates the list of Client Certs will be tried based off some property of the cert such as common name or thumbprint.  Should expand the doc here to explain why it's `X509Certificate2Collection`

Reply:
Also - should this callback return a [SslStreamCertificateContext](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslstreamcertificatecontext.create?view=net-5.0) instead of an `X509Certificate2Collection`, since it appears that is the correct API for building a client certificate chain with intermediates

Reply:
what about `X509ChainPolicy` it needs that as well

Reply:
Hard for me to weigh in on this with my limited vision of customers beyond my organization (and what they need).  The most extreme and simple would be to expose the hook for `SslClientAuthenticationOptions` or for a `Func<SslClientAuthenticationOptions,SslClientAuthenticationOptions>` which allows for customizable overrides of the default built in `SslStreamConnection`.  Between that and the current state just seems like different levels of tradeoff between flexibility and consistency.

Reply:
ok, trying the tests without `CertificateChainPolicy` doesn't seem to make any difference. Also looks like it might mess up the OCPS (I might've remembered the acronym wrong) feature:

> If not null, [CertificateRevocationCheckMode](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslclientauthenticationoptions.certificaterevocationcheckmode?view=net-8.0#system-net-security-sslclientauthenticationoptions-certificaterevocationcheckmode) and [SslCertificateTrust](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslcertificatetrust?view=net-8.0) are ignored.

Reply:
> Also - should this callback return a [SslStreamCertificateContext](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslstreamcertificatecontext.create?view=net-5.0) instead of an X509Certificate2Collection, since it appears that is the correct API for building a client certificate chain with intermediates

the only problem is that `ClientCertificateContext` option (which gets assigned) is not available in net6.0

Reply:
Yes, that API popped up in net8, there is a mega thread about it here: https://github.com/dotnet/runtime/issues/26323